drivers/pwm/pwm_nrfx.c

```diff
@@ -32,36 +32,14 @@ struct pwm_nrfx_config {
 struct pwm_nrfx_data {
 	uint32_t period_cycles;
 	uint16_t seq_values[NRF_PWM_CHANNEL_COUNT];
+	/* Bit mask indicating channels that need the PWM generation. */
+	uint8_t  pwm_needed;
 	uint8_t  prescaler;
 	uint8_t  initially_inverted;
 	bool     stop_requested;
 };
 
```

Comment:
maybe add something like 
`BUILD_ASSERT(NRF_PWM_CHANNEL_COUNT <= 8, "Implementation limited to 8 channels")`

Reply:
Added.